src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -978,6 +981,91 @@ export class ResultsViewPageStore {
         }
     });
 
+
+    readonly filteredAlterationsDataByUnflattenedOQLLine = remoteData<
+    IQueriedMergedTrackCaseData[]
+>({
+    await: () => [
+        this.filteredAlterations,
+        this.annotatedMolecularData,
+        this.selectedMolecularProfiles,
+        this.defaultOQLQuery,
+        this.samples,
+        this.patients
+    ],
+    invoke: () => {
+        const data = [...(this.filteredAlterations.result!), ...(this.annotatedMolecularData.result!)];
+        const accessorsInstance = new accessors(this.selectedMolecularProfiles.result!);
+        const defaultOQLQuery = this.defaultOQLQuery.result!;
+        const samples = this.samples.result!;
+        const patients = this.patients.result!;
+
+        if (this.oqlQuery.trim() === '') {
+            return Promise.resolve([]);
+        } else {
+            const filteredAlterationsByOQLLine: UnflattenedOQLLineFilterOutput<AnnotatedExtendedAlteration>[] = (
+                filterCBioPortalWebServiceDataByUnflattenedOQLLine(
+                    this.oqlQuery,
+                    data,
+                    accessorsInstance,
+                    defaultOQLQuery
+                )
+            );
+
+            return Promise.resolve(filteredAlterationsByOQLLine.map(
+                (oql) => ({
+                    cases: groupDataByCase(oql, samples, patients),
+                    oql,
+                    list: filterSubQueryData(
+                        oql, defaultOQLQuery,
+                        data, accessorsInstance,
+                        samples, patients
+                    )
+                })
+            ));
+        }
+    }
+});
+
+    readonly isSampleAlteredMap = remoteData({
+        await: () => [this.filteredAlterationsDataByUnflattenedOQLLine, this.samples],
+        invoke: async() => {          
+            const result : {[x: string]: boolean[]} = {};
+            this.filteredAlterationsDataByUnflattenedOQLLine.result!.forEach((element, key) => {
+                //1: is not group(1.1, 1.2)
```

Comment:
not sure what 1.1, 1.2 means.

Reply:
Based on the user input query. There is two main type of query: Gene(like EGFR / EGFR: MUT) and Gene group(like ["RAF" BRAF RAF1 ARAF]). I am using 1.1 to represent is gene like EGFR, which don't need to trim the last character of the oqlString. The 1.2 represents a gene with mutation type like EGFR: MUT, which need to remove the last punctuation from the oqlString.